### ISSUE TITLE
[Pylint] Guidelines checker should ignore `return` warnings for methods with `NoReturn`

### ISSUE THREAD

**Issue by mccoyp** [Azure/azure-sdk-for-python#32318]
A function can use the [`typing.NoReturn`](https://docs.python.org/3/library/typing.html#typing.NoReturn) type to indicate that it isn't implemented or doesn't `return` anything. Here's a recent example from Key Vault, where a method just raises a `NotImplementedError` if called:
https://github.com/Azure/azure-sdk-for-python/blob/3b50b96918832dc2cf2ee02276bd3b6c5987c954/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py#L169-L173
If possible, the pylint guideline checker should skip raising `return`-related warnings -- e.g. `docstring-missing-return` and `docstring-missing-rtype` -- if the return type is `NoReturn`.

**Comment by rahuja23** [comment]
@l0lawrence  Is this up for grabs?

**Comment by l0lawrence** [comment]
Yes go for it -- the code is under https://github.com/Azure/azure-sdk-tools/tree/main/tools/pylint-extensions/azure-pylint-guidelines-checker

**Comment by rahuja23** [comment]
@l0lawrence  I added a PR for the issue please let me know what you think 😄 : https://github.com/Azure/azure-sdk-tools/pull/10695
